### PR TITLE
Update shard.yml metadata

### PR DESCRIPTION
**Pull request by syeopite**
Changes are mostly based off of the Crystal compiler's own shard.yml

**Remember to bump the version attribute when creating a release!**

### CODE REVIEW

**Review by unixfox — APPROVED** [review]

File: shard.yml [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+version: 2.20240825.2-dev
 
 authors:
   - Omar Roth <omarroth@protonmail.com>
-  - Invidious team
+  - Invidious team <contact@invidious.io>
 
-targets:
-  invidious:
-    main: src/invidious.cr
```
Comment (SamantazFox) [inline shard.yml:10]: Why removing that, though?

File: shard.yml [hunk 2]
```diff
@@ -1,13 +1,12 @@
 name: invidious
-version: 0.20.1
+version: 2.20240825.2-dev
```
Comment (SamantazFox) [inline shard.yml:2]: How should we handle the `-dev`, in term of the release process? Do we remove it for the release commit, then add a new commit right after that that adds back `-dev`, like so?
```
ac4596fb (HEAD) Prepare for next version
def09876 (v2.2024MMDD.0) Release v2.2024MMDD.0
b1f8ee56 Merge pull request #9876
82f1f891 Fix stuff
```

File: shard.yml (hunk 1 above)
Comment (syeopite) [reply shard.yml:10]: I just modeled the file after the Crystal compiler one which doesn't have the target section.

Plus Invidious isn't intended to be built via shards

File: shard.yml (hunk 2 above)
Comment (syeopite) [reply shard.yml:2]: Yeah that's pretty much the only option.

In the Crystal compiler for example, they also just remove and readd the -dev prefix before and after each release https://github.com/crystal-lang/crystal/commits/master/shard.yml

File: shard.yml [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
 
 authors:
   - Omar Roth <omarroth@protonmail.com>
-  - Invidious team
+  - Invidious team <contact@invidious.io>
```
Comment (syeopite) [inline shard.yml:6]: Regarding the authors section should we change it to only include the Invidious team as to reflect the current authors?

Comment (SamantazFox) [reply shard.yml:6]: Why not, though `Invidious team and contributors` is more representative of all the people that worked on invidious!

**Review by SamantazFox — APPROVED** [review]

**Pull request merged**